ui/src/app/shared/components/search-bar.tsx

```diff
@@ -0,0 +1,99 @@
+import * as models from '../models';
+import * as React from 'react';
+import {Key, KeybindingContext} from 'argo-ui/v2';
+import {ContextApis} from '../context';
+import {Autocomplete} from 'argo-ui';
+
+function tryJsonParse(input: string) {
+    try {
+        return (input && JSON.parse(input)) || null;
+    } catch {
+        return null;
+    }
+}
+
+export const SearchBar = (props: {content: string; ctx: ContextApis; apps: models.Application[]; currentApp?: string}) => {
+    const {content, ctx, apps, currentApp} = {...props};
+    const appList = apps.map(app => (currentApp === app.metadata.name ? `${app.metadata.name} (current app)` : app.metadata.name));
```

Comment:
One small nitpick....perhaps put `${currentApp} (current app)` in a const after line 16, and then on lines 17 (after the ?) and 90, you can refer to this const.  That way, if we need to change the string, we do it in one place.

Reply:
Good point, I will make another push for this update. Thank you!

Reply:
Hey @ciiay , this change looks ok.  I think you can go ahead and ask a maintainer to approve/merge.